grpc/src/test/java/com/linecorp/armeria/server/grpc/GrpcHealthCheckServiceTest.java

```diff
@@ -0,0 +1,363 @@
+                server.httpUri(GrpcSerializationFormats.PROTO),
+                HealthBlockingStub.class);
+        HealthCheckRequest request = HealthCheckRequest.getDefaultInstance();
+        HealthCheckResponse response = client.check(request);
+        assertThat(response.getStatus()).isEqualTo(ServingStatus.SERVING);
+
+        request = HealthCheckRequest.newBuilder()
+                                    .setService("com.linecorp.armeria.grpc.testing.TestService")
+                                    .build();
+        response = client.check(request);
+        assertThat(response.getStatus()).isEqualTo(ServingStatus.SERVING);
+
+        final StatusRuntimeException exception = assertThrows(StatusRuntimeException.class, () -> client.check(
+                HealthCheckRequest.newBuilder()
+                                  .setService("com.linecorp.armeria.grpc.testing.NotFoundTestService")
+                                  .build()));
+        assertThat(exception.getStatus().getCode()).isEqualTo(Status.NOT_FOUND.getCode());
+        assertThat(exception.getMessage()).isEqualTo(
+                "NOT_FOUND: The service name(com.linecorp.armeria.grpc.testing.NotFoundTestService) " +
+                "is not registered in this service");
+    }
+
+    @Test
+    void watch() throws Exception {
+        final HealthCheckRequest request = HealthCheckRequest.getDefaultInstance();
+        final HealthStub client = Clients.newClient(
+                server.httpUri(GrpcSerializationFormats.PROTO),
+                HealthStub.class);
+        final StreamRecorder<HealthCheckResponse> recorder = StreamRecorder.create();
+        client.watch(request, recorder);
+        TimeUnit.SECONDS.sleep(1);
+        serveHealth.setHealthy(false);
+        TimeUnit.SECONDS.sleep(1);
+        serveHealth.setHealthy(true);
+        TimeUnit.SECONDS.sleep(1);
+        recorder.onCompleted();
+        final List<ServingStatus> responses = recorder.getValues()
+                                                      .stream()
+                                                      .map(HealthCheckResponse::getStatus)
+                                                      .collect(Collectors.toList());
+        assertThat(responses).containsExactlyElementsOf(
+                ImmutableList.of(ServingStatus.SERVING, ServingStatus.NOT_SERVING, ServingStatus.SERVING));
+    }
+
+    @Test
+    void watchTimeoutDisableCheck() throws Exception {
+        final HealthCheckRequest request = HealthCheckRequest.getDefaultInstance();
+        final HealthStub client = Clients.builder(server.httpUri(GrpcSerializationFormats.PROTO))
+                                         .responseTimeoutMillis(0)
+                                         .build(HealthStub.class);
+        final StreamRecorder<HealthCheckResponse> recorder = StreamRecorder.create();
+        client.watch(request, recorder);
+        TimeUnit.SECONDS.sleep(20);
```

Comment:
I think 20 seconds is too big to wait for finishing the test. 😄 It will increase the total build time so we need to decrease this value.
How about changing the value of `requestTimeout` to 3 seconds https://github.com/line/armeria/pull/3963/files#r765502796
and changing this value to 5 seconds? 😄

Reply:
It seems good.
I'll change them.